tarn/digest.py

```diff
@@ -27,14 +27,16 @@ def key_to_relative(key: AnyStr, levels: Sequence[int]):
         key = key.hex()
 
     # TODO: too expensive?
-    assert len(key) == get_digest_size(levels, string=True), (len(key), get_digest_size(levels, string=True))
+    assert len(key) >= get_digest_size(levels, string=True), (len(key), get_digest_size(levels, string=True))
```

Comment:
mb we should allow the last value of levels to be `-1` or `None`? this might make the structure simpler to reason about